src/DynExpManager/Instruments/NP_Conex_CC.cpp

```diff
@@ -0,0 +1,390 @@
+		auto Owner = DynExp::dynamic_Object_cast<NP_Conex_CC>(&Instance.GetOwner());
+
+		*InstrData->HardwareAdapter << Util::ToStr(InstrParams->ConexAddress.Get()) + "PR" + Util::ToStr(Position / Owner->GetFloatToIntConversion());
+
+		return {};
+	}
+
+	DynExp::TaskResultType NP_Conex_CC_Tasks::StopMotionTask::RunChild(DynExp::InstrumentInstance& Instance)
+	{
+		auto InstrParams = DynExp::dynamic_Params_cast<NP_Conex_CC>(Instance.ParamsGetter());
+		auto InstrData = DynExp::dynamic_InstrumentData_cast<NP_Conex_CC>(Instance.InstrumentDataGetter());
+
+		// It is not possible to use an if-condition, since the update of the Conex_CCStatus is to slow.
+		*InstrData->HardwareAdapter << Util::ToStr(InstrParams->ConexAddress.Get()) + "ST";
+
+		return DynExp::TaskResultType();
+	}
+
+	void NP_Conex_CCStageData::ResetImpl(dispatch_tag<PositionerStageData>)
+	{
+		Conex_CCStatus.Set(0);
+		ErrorCode = NoError;
+		NumFailedStatusUpdateAttempts = 0;
+
+		ResetImpl(dispatch_tag<NP_Conex_CCStageData>());
+	}
+
+	bool NP_Conex_CCStageData::IsMovingChild() const noexcept
+	{
+		return Conex_CCStatus.Moving() || Conex_CCStatus.Homing();
+	}
+
+	bool NP_Conex_CCStageData::HasArrivedChild() const noexcept
+	{
+		return Conex_CCStatus.ReadyFromHoming() || Conex_CCStatus.ReadyFromMoving() || Conex_CCStatus.ReadyFromDisable()
+			|| Conex_CCStatus.ReadyTFromReady() || Conex_CCStatus.ReadyTFromTracking() || Conex_CCStatus.ReadyTFromDisableT();
+	}
+
+	bool NP_Conex_CCStageData::HasFailedChild() const noexcept
+	{
+		return ErrorCode != 0
+			|| Conex_CCStatus.NotReferencedFromReset() || Conex_CCStatus.NotReferencedFromHoming() || Conex_CCStatus.NotReferencedFromConfiguration()
+			|| Conex_CCStatus.NotReferencedFromDisable() || Conex_CCStatus.NotReferencedFromReady() || Conex_CCStatus.NotReferencedFromMoving()
+			|| Conex_CCStatus.NotReferencedNoParams() || Conex_CCStatus.Configuration() || Conex_CCStatus.DisableFromReady()
+			|| Conex_CCStatus.DisableFromMoving() || Conex_CCStatus.DisableFromTracking() || Conex_CCStatus.DisableFromReadyT()
+			|| Conex_CCStatus.TrackingFromReadyT() || Conex_CCStatus.TrackingFromTracking();
+	}
+
+	std::string NP_Conex_CC::AnswerToNumberString(std::string&& Answer, const char* StartCode)
+	{
+		auto Pos = Answer.find(StartCode);
+
+		if (Answer.empty())
+			throw Util::InvalidDataException("Received an empty answer.");
+		else if (Pos == std::string::npos)
+			throw Util::InvalidDataException("Received an unexpected answer.");
+
+		return Answer.substr(Pos + 2);
+	}
+
+	NP_Conex_CC::NP_Conex_CC(const std::thread::id OwnerThreadID, DynExp::ParamsBasePtrType&& Params)
+		: PositionerStage(OwnerThreadID, std::move(Params))
+	{
+	}
+
+	void NP_Conex_CC::OnErrorChild() const
+	{
+		auto InstrData = DynExp::dynamic_InstrumentData_cast<NP_Conex_CC>(GetInstrumentData());
+
+		*InstrData->HardwareAdapter << "ST"; // Stop movement on all controllers
```

Comment:
Why not only on the controller this instrument refers to? I would prefer that if there is no strong reason. Otherwise, other instruments might be affected when this instrument fails?